python/cudf/cudf/core/frame.py

```diff
@@ -1753,11 +1754,17 @@ def sample(
                     "weights is not yet supported for axis=0/index"
                 )
 
-            seed = (
-                np.random.randint(np.iinfo(np.int64).max, dtype=np.int64)
-                if random_state is None
-                else np.int64(random_state)
-            )
+            if random_state is None:
+                seed = np.random.randint(
+                    np.iinfo(np.int64).max, dtype=np.int64
+                )
+            elif isinstance(random_state, np.random.mtrand.RandomState):
+                pos = random_state._bit_generator.state["state"]["pos"]
+                seed = random_state._bit_generator.state["state"]["key"][
+                    0 if pos == 624 else pos
```

Comment:
In the comments, Could we have some description of why we are using `624` here or mention this reference: https://stackoverflow.com/a/32172816 above?